### ISSUE TITLE
ResourcePicker non functional. (was AppProvider breaks Page)

### ISSUE THREAD

**Issue by BarryCarlyon** [Shopify/polaris-react#649]
# Issue summary

Under Polaris v3.1.1
An AppProvider with a apiKey argument stops Page actions and titles "working"

## Expected behavior

A Page with a title and primary or secondary actions, should show the title and primary secondary actions when the AppProvider has an apiKey argument

## Actual behavior

Page Title and Action Buttons do not appear

```<AppProvider apiKey={apiKey}>```

![image](https://user-images.githubusercontent.com/20999/48769300-3a869600-ecb3-11e8-861a-97684701b31a.png)

```<AppProvider>```

![image](https://user-images.githubusercontent.com/20999/48769321-47a38500-ecb3-11e8-9fc5-29f9f2b060a9.png)

## Steps to reproduce the problem

Give AppProvider an apiKey. apiKey does not need to be valid.

> apiKey="cake"

Will cause the issue also

## Reduced test case

```
import * as React from 'react';
import { render } from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import App from './App';

import '@shopify/polaris/styles.css';
import { AppProvider } from '@shopify/polaris';

function renderApp() {
    const { apiKey } = window;

    render(
        <AppContainer>
            <AppProvider apiKey={apiKey}>
                    <Page
                        title="What would you like to do today?"
                        primaryAction={{
                            content: 'Index',
                            onAction: () => {
                                //snip
                            }
                        }}
                        secondaryActions={
                            [
                                {
                                    content: 'Index',
                                    onAction: () => {
                                       // snip
                                    }
                                }
                            ]
                        }
                    >
                        <p>Test</p>
                    </Page>
            </AppProvider>
        </AppContainer>,
        document.getElementById('root')
    );
}

renderApp();

if (module.hot) {
    module.hot.accept();
}
```

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.1.1 (present also in 3.1.0)
- Browser: Chrome 70.0.3538.102 
- Device: Mac
- Operating System: Sierra 10.12.6

**Comment by dpersing** [comment]
Hi @BarryCarlyon! The Polaris team is looking into this. We'll let you know once we know more.

**Comment by danrosenthal** [comment]
@BarryCarlyon, from what I can glean from this issue, it appears that your problem with Page is expected, and not, in fact, a bug. It is the result of us not being clear enough in documentation. I've got a [work in progress PR](https://github.com/Shopify/polaris-react/pull/653) up to address that.

So the issue is that we have combined the former `EmbeddedPage` component into the existing `Page` component. When you pass an API key to `AppProvider`, that acts as a signal to tell `Page` to delegate to the Shopify App Bridge. So what you're seeing is `Page` rendering as intended in an embedded app context.

Regarding your problem with ResourceList, could you please expand on what you mean by broken? What isn't working for you?

**Comment by BarryCarlyon** [comment]
So there is no method to put buttons in the top bar any more?

I will need to rebuild my entire navigation.

Sorry I mean `ResourcePicker` but wrote list instead

`ResourcePicker` will not open under Polaris 3.

```
                                    <ResourcePicker
                                        resourceType="Collection"
                                        allowMultiple
                                        open={this.state.openCollectionPicker}
                                        onCancel={() => this.setState({openCollectionPicker: false})}
                                        onSelection={(resources) => {
                                            this.setState({openCollectionPicker: false});
                                            //snip
                                        }}
                                    />
```

**Comment by danrosenthal** [comment]
@markiiikram would you mind taking a look?

**Comment by BarryCarlyon** [comment]
Yup I switched to "ResourcePicker" from '@Shopify/polaris'
I don't use it's title anyway
And I'm using `resourceType="Collection"` as I'm after the collections rather than products

as documented at https://polaris.shopify.com/components/overlays/resource-picker

I will take a look at Page/Navigation.

**Comment by markiiikram** [comment]
Hey @BarryCarlyon, I believe we may have made a typo. Can you please try `import { ResourcePicker } from '@shopify/polaris';` (lowercase S on Shopify). If that's not the issue, can you let me know if 
```
      <ResourcePicker
        resourceType="Product"
```
works for you?

**Comment by BarryCarlyon** [comment]
```
import { ResourcePicker, TextField, Checkbox, Select, Button, Tooltip } from '@shopify/polaris';

// snip

                                    <ResourcePicker
                                        resourceType="Collection"
                                        allowMultiple
                                        open={true}
                                        onCancel={() => this.setState({openCollectionPicker: false})}
                                        onSelection={(resources) => {
                                            this.setState({openCollectionPicker: false});
                                        }}
                                    />

```

A change to

```
                                    <ResourcePicker
                                        resourceType="Product"
                                        allowMultiple
                                        open={true}
                                        onCancel={() => this.setState({openCollectionPicker: false})}
                                        onSelection={(resources) => {
                                            this.setState({openCollectionPicker: false});
                                        }}
                                    />
```

same result. No picker

**Comment by BarryCarlyon** [comment]
```
Hi @BarryCarlyon , Yes, you can still put buttons in the "top bar" for embedded apps. See the screenshot below, they now appear in a second row below the App name and your company name.

screen shot 2018-11-14 at 10 26 23 am

As Dan said, include a valid API Key in the AppProvider apiKey prop, and use the Page component.
```

That doesn't work:
```
                    <Page
                        title="Products"
                        secondaryActions={
                            [
                                {
                                    content: 'Index',
                                    onAction: () => {
                                        this.props.resetView()
                                    }
                                }
                            ]
                        }
                    >
```

I have no button… Which was the original issue of this thread…

**Comment by BarryCarlyon** [comment]
![image](https://user-images.githubusercontent.com/20999/48937442-00610280-ef06-11e8-8ee3-975d7171ea06.png)

Same code.

Left is live with Polaris 2x right is Dev with Polaris 3x

The `<Page` does nothing

**Comment by dfmcphee** [comment]
Hey @BarryCarlyon, are you still using the legacy EASDK anywhere in your code outside of Polaris? 

We saw another issue with similar symptoms to this recently and it turns out they were using both the new App Bridge in Polaris 3 and the legacy EASDK, which was interfering with the App Bridge setup.

In that case they were calling `ShopifyApp.init` somewhere in their app. Any chance you have the same?

**Comment by BarryCarlyon** [comment]
Nope. This is a pure react/Polaris app

No need to include EASDK if you/Polaris is doing it all for me.

**Comment by markiiikram** [comment]
@BarryCarlyon Are you doing server-side rendering?

**Comment by michelleyschen** [comment]
Also related.
cc @davekilljoy, @markiiikram does either of you have historical context on why the title in embedded app should be in the top bar rather than in page?

**Comment by davekilljoy** [comment]
@katbuilt would have the most context on the embedded app top bar.

**Comment by orbanbotond** [comment]
I confirm that neither Page nor Resource pickers shows... :(

**Comment by dpersing** [comment]
Hello all! A fix for this has been as part of version 3.2.1 ([see the release info for details](https://github.com/Shopify/polaris-react/releases/tag/v3.2.1)). I'll close this issue, but please let us know if problems persist.
